pkg/kinflate/util/interface.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Transformer interface {
+	// Transform takes a map GroupVersionKindName->Unstructured and do
+	// some transformation (e.g. prefix name or apply additional labels) on
+	// the objects in the same map.
+	Transform(m map[GroupVersionKindName]*unstructured.Unstructured) error
+	// TransformBytes is a wrapper of func Transform.
+	// It takes a byte array, converts it into a map GroupVersionKindName->Unstructured.
+	// Then call Transform to do the transformation.
+	// It returns a byte array of encoded transformed objects.
+	TransformBytes([]byte) ([]byte, error)
+}
```

Comment:
this file should be called transformer.go

An interface cannot commit itself to be a wrapper of another interface or to be implemented in a particular way.

Suggest one interface:
```
type Transformer interface {
  // Transform modifies objects in a map, e.g. add prefixes or additional labels.
  Transform(m map[GroupVersionKindName]*unstructured.Unstructured) error
}
``

because this interface:
```
type Transformer interface {
  // Transform converts an array of bytes into a new array of bytes.
  Transform([]byte) ([]byte, error)
}
```
is too general - it describes every program ever written.  

Besides this seems to be going down a road where there are many different implementations of the GVKM map transformer, but only one implementation of the []byte to []byte transformer (which will marshal, call transform, then unmarshal)